go/adcm/status/host.go

```diff
@@ -0,0 +1,169 @@
+package status
+
+type Host struct {
+	Id              int  `json:"id"`
+	MaintenanceMode bool `json:"maintenance_mode"`
+}
+
+type dbHost map[int]Host
+
+type hostRequest struct {
+	command          string
+	id               int
+	maintenance_mode bool
+}
+
+type hostResponse struct {
+	code int
+	ok   bool
+	host Host
+}
+
+type HostStorage struct {
+	in    chan hostRequest
+	out   chan hostResponse
+	db    dbHost
+	label string
+}
+
+func newHostStorage(db dbHost, label string) *HostStorage {
+	return &HostStorage{
+		in:    make(chan hostRequest),
+		out:   make(chan hostResponse),
+		db:    db,
+		label: label,
+	}
+}
+
+func (hs *HostStorage) run() {
+	logg.I.f("start storage %s server", hs.label)
+	for {
+		request := <-hs.in
+		logg.I.f("Storage %s command: %+v", hs.label, request.command)
+		switch request.command {
+		case "retrieve":
+			value, ok := hs.db.retrieve(request.id)
+			hs.out <- hostResponse{host: value, ok: ok}
```

Comment:
Maybe call an update if not `ok`? Or creating new object each time we have no info on a host is okay in this case?
When don't we know about the host anyway?